feature-libs/pdf-invoices/components/invoices-list/invoices-list.component.spec.ts

```diff
@@ -0,0 +1,462 @@
+  let component: InvoicesListComponent;
+  let fixture: ComponentFixture<InvoicesListComponent>;
+  let pdfInvoicesFacade: PDFInvoicesFacade;
+  let translationService: TranslationService;
+  let downloadService: FileDownloadService;
+  let globalMessageService: GlobalMessageService;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [I18nTestingModule, IconTestingModule],
+      declarations: [
+        InvoicesListComponent,
+        MockPaginationComponent,
+        MockSortingComponent,
+      ],
+      providers: [
+        { provide: PDFInvoicesFacade, useClass: MockPDFInvoicesFacade },
+        { provide: FileDownloadService, useClass: MockFileDownloadService },
+        { provide: LanguageService, useClass: MockLanguageService },
+        { provide: TranslationService, useClass: MockTranslationService },
+        { provide: GlobalMessageService, useClass: MockGlobalMessageService },
+      ],
+    }).compileComponents();
+    pdfInvoicesFacade = TestBed.inject(PDFInvoicesFacade);
+    translationService = TestBed.inject(TranslationService);
+    downloadService = TestBed.inject(FileDownloadService);
+    globalMessageService = TestBed.inject(GlobalMessageService);
+
+    spyOn(globalMessageService, 'add').and.callThrough();
+    spyOn(translationService, 'translate').and.returnValue(of('test'));
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(InvoicesListComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should show feature not enabled error when the API returns error', () => {
+    spyOn(pdfInvoicesFacade, 'getInvoicesForOrder').and.returnValue(
+      throwError(mockInvoicesNotEnabledError)
+    );
+    spyOn(component, 'getNotEnabledError').and.callThrough();
+
+    // Change the page
+    const newPage = 3;
+    component.pageChange(newPage);
+    expect(component.getNotEnabledError).toHaveBeenCalled();
+    expect(globalMessageService.add).toHaveBeenCalledWith(
+      { key: 'pdfInvoices.featureNotEnabled' },
+      GlobalMessageType.MSG_TYPE_ERROR
+    );
+  });
+
+  it('should read document list', () => {
+    let orderInvoiceList: OrderInvoiceList = {};
+    let queryParams: InvoiceQueryParams = {};
+    component.invoicesList$
+      .pipe(take(1))
+      .subscribe((value: OrderInvoiceList) => (orderInvoiceList = value));
+    expect(orderInvoiceList).toEqual(mockOrderInvoiceList);
+    expect(component.pagination).toEqual({
+      currentPage: 0,
+      pageSize: 5,
+      totalPages: 4,
+      totalResults: 16,
+      sort: 'invoiceId:asc',
+    });
+    component.queryParams$
+      .pipe(take(1))
+      .subscribe((value: InvoiceQueryParams) => (queryParams = value));
+    expect(queryParams).toEqual({
+      currentPage: 0,
+      pageSize: 5,
+      fields: InvoicesFields.FULL,
+      sort: 'invoiceId:asc',
+    });
+  });
```

Comment:
```suggestion
  it('should read document list', (done) => {
    let orderInvoiceList: OrderInvoiceList = {};
    let queryParams: InvoiceQueryParams = {};
    component.invoicesList$
      .pipe(take(1))
      .subscribe((value: OrderInvoiceList) => 
      orderInvoiceList = value;
      done();
      );
    expect(orderInvoiceList).toEqual(mockOrderInvoiceList);
    expect(component.pagination).toEqual({
      currentPage: 0,
      pageSize: 5,
      totalPages: 4,
      totalResults: 16,
      sort: 'invoiceId:asc',
    });
    component.queryParams$
      .pipe(take(1))
      .subscribe((value: InvoiceQueryParams) => {
      queryParams = value; 
      done();
      });
    expect(queryParams).toEqual({
      currentPage: 0,
      pageSize: 5,
      fields: InvoicesFields.FULL,
      sort: 'invoiceId:asc',
    });
  });
```
Make similar change in other places where async functionality is called.

Reply:
noted, thanks Pawel